src/suricata.c

```diff
@@ -450,6 +454,27 @@ static void GlobalsDestroy(SCInstance *suri)
     suri->pid_filename = NULL;
 }
 
+/**
+ * \brief Used to send OS specific notification of running threads
+ *
+ * \retval TmEcode TM_ECODE_OK on success; TM_ECODE_FAILED on failure.
+ */
+static TmEcode OnNotifyRunning(void)
+{
+    SCLogNotice("All threads in running state.");
+
+#if HAVE_LIBSYSTEMD
+    if (sd_notify(0, "READY=1") < 0) {
+        SCLogWarning(SC_ERR_SYSCALL, "failed to notify systemd");
+        /* Please refer to:
+         * https://www.freedesktop.org/software/systemd/man/sd_notify.html#Return%20Value
+         * for discussion on why failure should not be considered an error */
+    }
+#endif
+
+    return TM_ECODE_OK;
```

Comment:
Yes you are correct. My thinking was to have it as a template for potential/future developments. But I suppose if it is refactored it can then be adapted accordingly. Thanks, I will update this.